File: Darts/src/burlton/dartzee/code/bean/X01ScoreRenderer.java
package burlton.dartzee.code.bean;

import java.awt.Color;

import burlton.desktopcore.code.bean.AbstractTableRenderer;
import burlton.dartzee.code.utils.DartsColour;

public class X01ScoreRenderer extends AbstractTableRenderer<Integer>
{
	@Override
	public Object getReplacementValue(Integer object)
	{
		return object;
	}
	
	@Override
	public void setCellColours(Integer score, boolean isSelected)
	{
		Color fg = DartsColour.getScorerForegroundColour(score);
		Color bg = DartsColour.getScorerBackgroundColour(score);
		
		if (isSelected)
		{
			fg = Color.WHITE;
			bg = DartsColour.getDarkenedColour(bg);
		}
		
		setForeground(fg);
		setBackground(bg);
	}
}

File: Darts/src/burlton/dartzee/code/bean/X01ScoreRenderer.kt
package burlton.dartzee.code.bean

import burlton.dartzee.code.utils.DartsColour
import burlton.dartzee.code.core.bean.AbstractTableRenderer
import java.awt.Color

class X01ScoreRenderer : AbstractTableRenderer<Int>()
{
    override fun getReplacementValue(value: Int) = value

    override fun setCellColours(typedValue: Int?, isSelected: Boolean)
    {
        val score = typedValue!!
        var fg = DartsColour.getScorerForegroundColour(score.toDouble())
        var bg = DartsColour.getScorerBackgroundColour(score.toDouble())

        if (isSelected)
        {
            fg = Color.WHITE
            bg = DartsColour.getDarkenedColour(bg)
        }

        foreground = fg
        background = bg
    }
}
